lwip/Makefile

```diff
@@ -0,0 +1,30 @@
+# Base library definitions
+TOCK_USERLAND_BASE_DIR ?= ..
+LIBNAME := lwip
+
+# This Makefile's directory, library is actually located in the `lwip`
+# submodule under this directory
+$(LIBNAME)_DIR := $(TOCK_USERLAND_BASE_DIR)/$(LIBNAME)
+
+# Include the filelist makefile of lwip. The filelist requires a variable named
+# `LWIPDIR` to be in scope, which points to the `src` directory inside the git
+# submodule. This is not to be confused with `$(LIBNAME)_DIR` (expands to
+# `lwip_DIR`), which is the top-level Tock directory name:
+LWIPDIR := $($(LIBNAME)_DIR)/$(LIBNAME)/src
+include $($(LIBNAME)_DIR)/lwip/src/Filelists.mk
```

Comment:
We discussed on the Network working group call moving this to include a vendored file, as the submodule might not be initialized yet by the time this line runs.